plugins/scaffolder-backend-module-confluence-to-markdown/src/actions/confluence/confluenceToMarkdown.examples.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TemplateExample } from '@backstage/plugin-scaffolder-node';
+import yaml from 'yaml';
+
+export const examples: TemplateExample[] = [
+  {
+    description:
+      'Downloads content from provided Confluence URLs and converts it to Markdown.',
+    example: yaml.stringify({
+      steps: [
```

Comment:
It can be nice to have a test for this, so that it continues to remain a valid example into the future. I have some examples of tests in the pull request here https://github.com/backstage/backstage/pull/18831

Reply:
I've added a test but for the life of me can't seem to get it to pass. Could you take a look, please?

Reply:
I was able to get past the most immediate issue by changing the readTree mock to this:

```
  const reader: UrlReader = {
    readUrl: jest.fn(),
    readTree: async () => ({dir: jest.fn()}),
    search: jest.fn(),
  };
```

I think you will also need to change the baseUrl at the top to this:

```
  const baseUrl = `https://confluence.example.com`;
```

Reply:
I was just demostrating that the change fixes the mocking. I was also getting type errors and didnt make any attempt to fix them. You would need to find the right syntax to use there.